### PR TITLE
WIP/HOLD: Switch to Web3 WebsocketProvider & rework sync_listener

### PR DESCRIPTION
**Pull request by pinkiebell**
Related #1364

### CODE REVIEW

**Comment by owocki** [comment]
is this WIP or ready for review?

File: app/dashboard/management/commands/sync_listener.py [hunk 1]
```diff
@@ -22,64 +22,107 @@
...
+        bounty = get_bounty(bounty_id, network)
+        did_change, old_bounty, new_bounty = web3_process_bounty(bounty)
+        if event == 'BountyActivated':
+            new_bounty.tx_hash_activated = tx_hash
+            new_bounty.save()
+            return True
+        if event == 'FulfillmentAccepted':
+            new_bounty.tx_hash_accepted = tx_hash
+            new_bounty.save()
+            return True
```
Comment (owocki) [inline app/dashboard/management/commands/sync_listener.py]: is it worth adding the other events to this list?

File: app/dashboard/migrations/0091_auto_20180712_1800.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('dashboard', '0090_auto_20180709_1525'),
+    ]
+
+    operations = [
```
Comment (owocki) [inline app/dashboard/migrations/0091_auto_20180712_1800.py]: i wonder if this should just be a json field with a list of many tx hashes... @mbeacom what do you think?

File: requirements/base.txt [hunk 3]
```diff
@@ -26,7 +26,7 @@ sendgrid
 slackclient
 eth-tester
 websockets
-web3==4.0.0b10
+web3==4.4.1
```
Comment (owocki) [inline requirements/base.txt]: might be worth a regression test of functionality elsewhere if were bumping 4 minor versions

File: app/dashboard/management/commands/sync_listener.py (hunk 1 above)
Comment (pinkiebell) [reply app/dashboard/management/commands/sync_listener.py]: @owocki 
Most of the time we have > Issued|Activated|ContributionAdded| > BountyFulfilled|BountyKilled > FulfillmentAccepted.

We can save all events and make those hash properties a list.

Comment (owocki) [reply app/dashboard/management/commands/sync_listener.py]: > Most of the time 

sure, but what about IncreasePayout event?  and the newFulfillment event?  those are pretty common too.

Comment (owocki) [reply app/dashboard/management/commands/sync_listener.py]: yes that makes sense to me

**Comment by owocki** [comment]
is this still WIP

**Comment by owocki** [comment]
also, how extensively has this been tested?  how long can it be expected to run without dying?

**Comment by pinkiebell** [comment]
I ran it several times for 2 - 11 hours. It works fine, except that infura is somtimes really slow or completely unresponsive, in this case; the socket will timeout and the code picks up from the last block.

**Comment by pinkiebell** [comment]
@owocki Let me know if you want me to decouple the tx_hashes & UI part from this PR.

**Comment by mbeacom** [comment]
Is this still a WIP @pinkiebell ? Can you resolve the merge conflicts?

**Comment by pinkiebell** [comment]
@owocki 
No WIP. Rebased 👍

**Comment by pinkiebell** [comment]
@mbeacom 
s/owocki/mbeacom

**Comment by pinkiebell** [comment]
This actually worked very well for a few weeks, but infura is too unreliable recently (esp. for event logs).
So I pushed it back to WIP/HOLD to improve things a bit/to cope with infura performance 👍

**Comment by owocki** [comment]
cc @mbeacom in case our web3 RPC is still up and running

**Comment by pinkiebell** [comment]
@mbeacom @owocki 
Any thoughts on running an own geth instance?

**Comment by mbeacom** [comment]
@pinkiebell We've been discussing this a bit and are tracking here:  https://github.com/gitcoinco/web/issues/2394

I don't think it'd be a problem. We were running our own for a while, but ultimately turned it off and reverted back to Infura because we no longer saw a need for it.  I still have our geth k8s charts and configuration files, so I could have one up relatively quickly...  It's just a matter of determining the TCO and cost/benefit of doing this.  I think we'd see a major performance boost with respect to sync times.

That said, we'd want to run more than a single node to ensure HA/redundancy/resilience for production use.

**Pull request closed without merging**